File: 2023/java/src/test/java/org/frieder/aoc/day7/TestDay7B.java
package org.frieder.aoc.day7;

import org.frieder.aoc.day7.b.Solution7B;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertEquals;

public class TestDay7B {
    @Test
    public void testExample() throws IOException {
        assertEquals(5905, Solution7B.getResult("../resources/7/test_a.txt"), 0);
    }

    @Test
    public void testResult() throws IOException {
        assertEquals(251003917, Solution7B.getResult("../resources/7/input.txt"), 0);
    }
}

File: 2023/php/test/TestDay7B.php
<?php declare(strict_types=1);

$loader = require __DIR__ . '/../vendor/autoload.php';

use day7\b\Solution7B;
use PHPUnit\Framework\TestCase;
use utils\TestLogger;

final class TestDay7B extends TestCase
{

    public function testExample(): void
    {
        $this->assertEqualsWithDelta(5905, Solution7B::getResult("../resources/7/test_b.txt", TestLogger::createTestLogger()), 0);
    }

    public function testResult(): void
    {
        $this->assertEqualsWithDelta(251003917, Solution7B::getResult("../resources/7/input.txt", TestLogger::createTestLogger()), 0);
    }

}